src/transformStateWithClones.js

```diff
@@ -7,7 +7,46 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let initialState = { ...state };
+  const statesArray = [];
+
+  for (const action of actions) {
+    reducer(action);
+  }
+
+  function reducer(action) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const key in action.extraData) {
+          initialState[key] = action.extraData[key];
+        }
+
+        statesArray.push({ ...initialState });
```

Comment:
checklist.md:
4. [DONT REPEAT YOURSELF] - If you perform same action in all switch cases - do it just once afterwards